### ISSUE TITLE
About the reset mask

### ISSUE THREAD

**Issue by 127161782** [aai-institute/tianshou#734]
- [ ] [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [ ] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [ ] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, gym, torch, numpy, sys
  print(tianshou.__version__, gym.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
  ```
I use preprocess_fn, in order to get the mask matrix after env reset, and I modified the reset_env function of Collector。
first，I modified training script：
```python
    def preprocess_fn(**kwargs):
        batch = Batch()
        batch.update(kwargs)
        if "rew" not in kwargs: #
            batch.info = {"mask": np.array(env.mask)}

            return batch
        return batch

```
The second step, modify the collector：
```python

    def reset_env(self) -> None:
        """Reset all of the environments."""
        obs = self.env.reset()
        if self.preprocess_fn:
            ret = self.preprocess_fn(obs=obs, env_id=np.arange(self.env_num))
            # obs = ret.get("obs", obs)
            # info = ret.get("info",Batch())
            self.data.update(ret)      # 主要修改
        self.data.obs = obs
```
my policy main function：
```python
    def forward(  # type: ignore
            self,
            batch: Batch,
            state: Optional[Union[dict, Batch, np.ndarray]] = None,
            input: str = "obs",
            **kwargs: Any,
    ) -> Batch:
        obs = batch[input]
        obs = obs.obs if hasattr(obs, "obs") else obs  # pre_fn,必哟有obs -->batch
        logits, hidden = self.actor(obs, state=state, info=batch.info)
        logits = logits.view(logits.shape[0], self.action_num, self.finger_num, -1)  #
        logits = self.mask_logits_act(logits, getattr(batch.info, "mask", None))  # 
        dist = Categorical(logits=logits)
        act = dist.sample()
        return Batch(logits=logits, act=act, state=hidden, dist=dist)

    def mask_logits_act(
            self, logits: torch.Tensor, mask: Optional[np.ndarray]
    ) -> torch.Tensor:
        """Compute the q value based on the network's raw output and action mask."""
        if mask is not None:
            logits = logits + torch.tensor(mask)  # logits torch.Size([20, 30, 3, 26])    mask (30, 3, 26)
            # exit()
        return logits
```
**logits = logits + torch.tensor(mask)  # logits torch.Size([20, 30, 3, 26])    mask (30, 3, 26)**
logits is inconsistent with mask dimension。
Don't know why the mask is not accumulated. Can you point me to the modification？

**Comment by Trinkle23897** [comment]
>         if "rew" not in kwargs: #
>            batch.info = {"mask": np.array(env.mask)}

In process_fn, the data has already batched. You use a single env's mask to overwrite batch data, that's obviously not a correct approach.

**Comment by 127161782** [comment]
OK, I got it, thanks. I changed it in the following way and it worked.
```python
    if "rew" not in kwargs: #
        batch.info = {"mask": np.array([env.mask]*len(batch))}
        return batch
```
